### PR TITLE
Rewrite CSS guidelines per Tailwind conversion

### PR DESCRIPTION
**Pull request by lyzadanger**
Update CSS guidelines for our current Tailwind setup. We can be briefer because Tailwind does some of the organizational work for us and we don't have to make so many decisions about organization and naming.

Part of https://github.com/hypothesis/frontend-shared/issues/311

### CODE REVIEW

File: docs/css-style-guide.md [hunk 1]
```diff
@@ -1,136 +1,48 @@
...
+  - Each application may extend that Tailwind preset with local values (e.g. [the client's current configuration](https://github.com/hypothesis/client/blob/master/tailwind.config.mjs))
+- Make use of layers:
+  - Each CSS (SASS) entrypoint should enable the [Tailwind `base`, `components` and `utilities` layers](https://tailwindcss.com/docs/functions-and-directives#layer) and should structure any partials/imports along the lines of those layers.
+  - Using Tailwind layers for now puts us in a prime position to take advantage of native CSS layers.
+- Be thoughtful when authoring external CSS:
+  - External CSS: it's sometimes necessary! We're not aiming for an airtight system in which there is no external CSS. Use your judgment. On the other hand, check first: can you do it with Tailwind/a utility class instead?
+  - Add external CSS to the relevant Tailwind layer, and use the [`@apply` directive](https://tailwindcss.com/docs/functions-and-directives#apply) to inline Tailwind utility classes.
+  - Be alert if you find yourself trying to come up with names for things. Re-consider if there is a Tailwind-y way to avoid naming and abstractions.
+- Don't rely too much on SASS:
```
Comment (robertknight) [inline docs/css-style-guide.md]: Although it is good to reduce the overall complexity of our tech stack, SASS has not been a problematic dependency, so if it still proves valuable, I wouldn't mind too much keeping it around for a while. The general advice not to rely heavily on advanced SASS features makes sense.

File: docs/css-style-guide.md [hunk 2]
```diff
@@ -1,136 +1,48 @@
...
-- Use color, typography and other values defined in the shared `variables` module
-  (or, even better: use a utility mixin) rather than hard-coding CSS rule values
-  in component CSS
-- We use flexbox instead for layout; there are some utility mixins and classes
-  to make component styling more convenient.
-- Avoid vendor prefixes ([autoprefixer](https://github.com/postcss/autoprefixer) will
-  add these automatically).
+Hypothesis is transitioning to a utility-first approach for CSS styling, supported by [Tailwind CSS](https://tailwindcss.com/). We use [SASS](https://sass-lang.com/) to build output CSS stylesheets.
+
+Using Tailwind's utility-first approach obviates the need to spend much time thinking about or documenting how things should be named or organized, and can prevent us from creating tempting but sometimes confusing or unmaintainable abstractions.
```
Comment (robertknight) [inline docs/css-style-guide.md]: Tailwind avoids the need to come up with labels for all the elements in a component that exist to provide style or structure, but of course many things in general still need thought about the naming (eg. entries in color or typography palettes, component names, etc.)

File: docs/css-style-guide.md [hunk 3]
```diff
@@ -1,136 +1,48 @@
...
+  - Each application may extend that Tailwind preset with local values (e.g. [the client's current configuration](https://github.com/hypothesis/client/blob/master/tailwind.config.mjs))
+- Make use of layers:
+  - Each CSS (SASS) entrypoint should enable the [Tailwind `base`, `components` and `utilities` layers](https://tailwindcss.com/docs/functions-and-directives#layer) and should structure any partials/imports along the lines of those layers.
+  - Using Tailwind layers for now puts us in a prime position to take advantage of native CSS layers.
+- Be thoughtful when authoring external CSS:
+  - External CSS: it's sometimes necessary! We're not aiming for an airtight system in which there is no external CSS. Use your judgment. On the other hand, check first: can you do it with Tailwind/a utility class instead?
+  - Add external CSS to the relevant Tailwind layer, and use the [`@apply` directive](https://tailwindcss.com/docs/functions-and-directives#apply) to inline Tailwind utility classes.
+  - Be alert if you find yourself trying to come up with names for things. Re-consider if there is a Tailwind-y way to avoid naming and abstractions.
```
Comment (robertknight) [inline docs/css-style-guide.md]: Specifically what _things_ are you referring to here?

**Review by robertknight — APPROVED** [review]
Let's get this guide updated based on our current plans. I suggest planning to revisit it in a few months with evolving conventions as we (well, mostly you!) gain more experience with Tailwind and as we get closer to completing out the transition from the BEM-ish approach.

I have some doubts in my mind about whether we will end up retiring SASS in the near future or if we will still find enough value in various elements of it. There are also some comments about avoiding the need for naming which just refers to naming of "things", but I think we you specifically mean class names for DOM elements that exist for styling or layout?

File: docs/css-style-guide.md (hunk 2 above)
Comment (lyzadanger) [reply docs/css-style-guide.md]: How about `s/things/CSS classes/`

File: docs/css-style-guide.md (hunk 3 above)
Comment (lyzadanger) [reply docs/css-style-guide.md]: Changing to "class names or Tailwind configuration property names"

File: docs/css-style-guide.md (hunk 1 above)
Comment (lyzadanger) [reply docs/css-style-guide.md]: Changing to:

```
- Don't rely too much on SASS' features:
  - Avoid SASS variables, mixins, and functions.
  - Remove unused SASS (variables, mixins, etc.) as you encounter them.
 ```

**Pull request merged**